Cabal/Distribution/PackageDescription/Configuration.hs

```diff
@@ -421,11 +421,14 @@ overallDependencies enabled (TargetSet targets) = mconcat depss
   where
     (depss, _) = unzip $ filter (removeDisabledSections . snd) targets
     removeDisabledSections :: PDTagged -> Bool
-    removeDisabledSections (Lib l)     = componentEnabled enabled (CLib l)
-    removeDisabledSections (SubLib _ l) = componentEnabled enabled (CLib l)
-    removeDisabledSections (Exe _ e)   = componentEnabled enabled (CExe e)
-    removeDisabledSections (Test _ t)  = componentEnabled enabled (CTest t)
-    removeDisabledSections (Bench _ b) = componentEnabled enabled (CBench b)
+    -- UGH. The embedded componentName in the 'Component's here is
+    -- BLANK.  I don't know whose fault this is but I'll use the tag
+    -- instead. -- ezyang
```

Comment:
Ins't it possible to fix the root of problem (blank names), then try to workaround it?